spec/syntax.md

```diff
@@ -0,0 +1,598 @@
+# DRAFT MessageFormat 2.0 Syntax
+
+## Table of Contents
+
+1. [Introduction](#introduction)
+   1. [Design Goals](#design-goals)
+   1. [Design Restrictions](#design-restrictions)
+1. [Overview & Examples](#overview--examples)
+   1. [Simple Messages](#simple-messages)
+   1. [Simple Placeholders](#simple-placeholders)
+   1. [Formatting Functions](#formatting-functions)
+   1. [Markup Elements](#markup-elements)
+   1. [Selection](#selection)
+   1. [Local Variables](#local-variables)
+   1. [Complex Messages](#complex-messages)
+1. [Comparison with ICU MessageFormat 1.0](#comparison-with-icu-messageformat-10)
+1. [Productions](#productions)
+   1. [Message](#message)
+   1. [Plain](#plain)
+   1. [Preamble](#preamble)
+   1. [Variants](#variants)
+   1. [Patterns](#patterns)
+   1. [Placeables](#placeables)
+   1. [Expressions](#expressions)
+   1. [Markup Elements](#markup-elements)
+1. [Tokens](#tokens)
+   1. [Text](#text)
+   1. [Names](#names)
+   1. [Quoted Strings](#quoted-strings)
+   1. [Escape Sequences](#escape-sequences)
+   1. [Whitespace](#whitespace)
+1. [Complete EBNF](#complete-ebnf)
+
+## Introduction
+
+This document defines the formal grammar describing the syntax of a single message.
+A separate syntax shall be specified to describe collections of messages (_MessageResources_),
+including message identifiers, metadata, comments, groups, etc.
+
+The document is part of the MessageFormat 2.0 specification,
+the successor to ICU MessageFormat, henceforth called ICU MessageFormat 1.0.
+
+### Design Goals
+
+The design goals of the syntax specification are as follows:
+
+1. The syntax should be an incremental update over the ICU MessageFormat 1.0 syntax
```

Comment:
The single message model in l10n has nothing to do with MF1, or with localization tooling.

This seems to conflate several independent issues:
- Translate 1 message at the time, independent of other messages. This is mostly dictated by localization frameworks. XLIFF and many localization tools have support for message grouping and references.
- Translate 1 source message to 1 target message (1:1 translation). This is a limitation in many localization tools, and currently gets in the way of proper support for plural (in MessageFormat, gettext, Android, others)
- The proposed syntax does not seem like an incremental update to MF1. In fact there are areas where it is clearly not functionally backward compatible (for examples select using "first match" instead of best match).

Reply:
@mihnita Could you clarify whether you're asking for some specific change in this design goal?

Reply:
I am asking for a full rewrite of bullet 1.
Since I don't even understand what it means, I can't suggest anything.
It connects completely unrelated issues, like _"we will paint this blue, for compatibility with Texas, and make it easier for people used to public transit"_
And insinuates (again) the baseless idea that we are sacrificing something for some fictitious MF 1 "incremental update"

Reply:
Here's how I suggest to rephrase the first bullet:

> The syntax should leverage the familiarity with ICU MessageFormat 1.0 in order to lower the barrier to entry and increase the chance of adoption. At the same time, the syntax should fix the [pain points of ICU MessageFormat 1.0](https://github.com/unicode-org/message-format-wg/blob/main/docs/why_mf_next.md).